packages/extension-base/src/services/keyring-service/context/handlers/Migration.ts

```diff
@@ -0,0 +1,249 @@
+    const UACanBeMigratedSortedByParent = this.sortUAByParent(UACanBeMigrated); // master account should be migrated before derived account
+    const migratedUnifiedAccountIds = await this.migrateUnifiedToUnifiedAccount(password, UACanBeMigratedSortedByParent);
+
+    // Get EligibleSoloAccounts
+    const rawSoloAccounts = this.getEligibleSoloAccounts(accountProxies);
+    const rawEligibleSoloAccounts = this.groupSoloAccountByPair(password, rawSoloAccounts);
+    const eligibleSoloAccountMap = this.accountProxiesToEligibleSoloAccountMap(rawEligibleSoloAccounts);
+
+    // Create UniqueId
+    const uniqueId = Date.now().toString();
+    const timeoutId = setTimeout(() => delete this.sessionIdToPassword[uniqueId], SESSION_TIMEOUT * 2);
+
+    this.sessionIdToPassword[uniqueId] = {
+      password,
+      timeoutId
+    };
+
+    return {
+      migratedUnifiedAccountIds,
+      soloAccounts: eligibleSoloAccountMap,
+      sessionId: uniqueId
+    };
+  }
+
+  public async migrateUnifiedToUnifiedAccount (password: string, accountProxies: AccountProxy[]): Promise<string[]> {
+    const unifiedAccountIds: string[] = [];
+    const modifiedPairs = structuredClone(this.state.modifyPairs);
+    let notMigrateDeriveBefore = true;
+
+    keyring.unlockKeyring(password);
+    this.parentService.updateKeyringState();
+
+    for (const unifiedAccount of accountProxies) {
+      const masterAccount = !unifiedAccount.parentId;
+      const proxyId = unifiedAccount.id;
+      const name = unifiedAccount.name;
+
+      if (masterAccount) {
+        const mnemonic = this.parentService.context.exportAccountProxyMnemonic({
+          password,
+          proxyId
+        }).result;
+
+        const accountChainTypes = unifiedAccount.chainTypes;
+        const newChainTypes = Object.values(AccountChainType).filter((type) => !accountChainTypes.includes(type) && SUPPORTED_ACCOUNT_CHAIN_TYPES.includes(type));
+        const keypairTypes = newChainTypes.map((chainType) => getDefaultKeypairTypeFromAccountChainType(chainType));
+
+        keypairTypes.forEach((type) => {
+          const suri = getSuri(mnemonic, type);
+          const pair = keyring.createFromUri(suri, {}, type);
+          const address = pair.address;
+
+          modifiedPairs[address] = { accountProxyId: proxyId, migrated: true, key: address };
+        });
+
+        this.state.upsertModifyPairs(modifiedPairs);
+        keypairTypes.forEach((type) => {
+          const suri = getSuri(mnemonic, type);
+          const { derivePath } = keyExtractSuri(suri);
+          const metadata = {
+            name,
+            derivationPath: derivePath ? derivePath.substring(1) : undefined
+          };
+
+          const rs = keyring.addUri(suri, metadata, type);
+          const address = rs.pair.address;
+
+          this.state._addAddressToAuthList(address, true);
+        });
+
+        unifiedAccountIds.push(proxyId);
+      } else { // derived account
```

Comment:
should split into another for loop

Reply:
will split the list `unifiedAccounts` into 2 list: `masterUnifiedAccounts` and `derivedUnifiedAccounts` and seperate the loop. This is just for improving code readibility.

Reply:
I've updated this ✅